src/drivers/imu/bosch/bmi088_i2c/BMI088_Accelerometer.cpp

```diff
@@ -908,11 +913,29 @@ bool BMI088_Accelerometer::NormalRead(const hrt_abstime &timestamp_sample)
 	const int16_t accel_y = combine(RATE_Y_MSB, RATE_Y_LSB);
 	const int16_t accel_z = combine(RATE_Z_MSB, RATE_Z_LSB);
 
+	if (accel_x == INT16_MIN) {
+		PX4_WARN("accel_x == INT16_MIN");
+	}
+
+	if (accel_y == INT16_MIN) {
+		PX4_WARN("accel_y == INT16_MIN");
+	}
+
+	if (accel_z == INT16_MIN) {
+		PX4_WARN("accel_z == INT16_MIN");
+	}
+
+	if (accel_x == INT16_MIN || accel_y == INT16_MIN || accel_z == INT16_MIN) {
```

Comment:
Is this described in the datasheet? What about INT16_MAX?

Reply:
INT16_MIN is binary "all ones" and seems to be a fluke in FIFO reading process. I didn't see anything about it in the datasheet, only some mentioning in forums. Direct reads from registers (opposed to FIFO) don't seem to produce those.